debug_module.tex

```diff
@@ -166,25 +166,29 @@ \section{Hart States}
 \section{Run Control} \label{runcontrol}
 
 For every hart, the Debug Module contains 4 conceptual bits of state: halt
-request, resume request, halt-on-reset request,  and hart reset.
-(The hart reset and halt-on-reset request bits are optional.)
-These bits all reset to 0. A debugger can write them for the currently selected
-harts through \Fhaltreq, \Fresumereq, \Fsetresethaltreq/\Fclrresethaltreq and
-\Fhartreset in \Rdmcontrol, only one of which may be high in any single write.
-In addition the DM receives halted, running, and resume ack signals from each
-hart.
-
-When a running hart receives a halt request, it responds by halting,
-deasserting its running signal, and asserting its halted signal. The halted
-signals of all selected harts are reflected in the \Fallhalted and \Fanyhalted
-bits. \Fhaltreq is ignored by halted harts.
-
-When a halted hart receives a resume request, it responds by resuming, clearing
-its halted signal, and asserting its running signal and resume ack signals. The
-resume ack signal is lowered when the resume request is deasserted.  These
-status signals of all selected harts are reflected in \Fallresumeack,
-\Fanyresumeack, \Fallrunning, and \Fanyrunning. \Fresumereq is ignored by
-running harts.
+request, resume ack, halt-on-reset request, and hart reset.  (The hart reset
```

Comment:
dont you mean resume request? Not resume ack?

Reply:
I do mean resume ack.

The DM needs to track resume ack, so it can zero it when resumereq is written 1.

Resume request is listed below as a "signal" because it's not level sensitive. When resumereq is written, the DM sends the hart a resume request, but it doesn't persist beyond that.

Does that make sense? How can I phrase this better?